### PR TITLE
feat(ksyms): required ksym scanning

### PR DESCRIPTION
**Pull request by NDStrahilevitz**
### 1. Explain what the PR does

f92253630 **chore(go.mod): update libbpfgo helpers**
0d01802e1 **feat(ksymbols): restore lazy ksyms implementtion**

```
Use modified ksymbols implementation. The new implementation may take
a list of required symbols and addresses to track. If the list is given,
symbol scanning will only save those symbols or addresses which were
given in the list. If a new symbol is queried, then a rescan is needed.

Refactor tracee initialization to find all necessary symbols to track
ahead of runtime.
```

### 2. Explain how to test it

No change in behavior, all previous tests should pass.
Memory should be reduced, testable with:
`ps -p `pgrep tracee` -o pid,vsz,rss,cmd`
and check rss.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -180,3 +180,5 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	kernel.org/pub/linux/libs/security/libcap/psx v1.2.68 // indirect
 )
+
+replace github.com/aquasecurity/libbpfgo/helpers => ../libbpfgo/helpers
```
Comment (NDStrahilevitz) [inline go.mod:184]: TODO: split PR and remove this

**Review by geyslan — COMMENTED** [review]
I did a cursory review, but didn't tested yet. Put some questions.

There's this error in the tests: https://github.com/aquasecurity/tracee/actions/runs/9290348284/job/25574577060?pr=4088#step:5:1471

**Comment by yanivagman** [comment]
To be replaced by #4092 that uses `environment` package instead of `libbpfgo/helpers`

File: pkg/ebpf/probes/trace.go [hunk 2]
```diff
@@ -22,8 +23,23 @@ const (
...
+func (t ProbeType) String() string {
+	switch t {
+	case KProbe:
+		return "kprobe"
+	case KretProbe:
+		return "kretprobe"
+	case Tracepoint:
+		return "tracepoint"
+	case RawTracepoint:
+		return "raw_tracepoint"
```
[the remark this answers is not in the document]
Comment (geyslan) [reply pkg/ebpf/probes/trace.go:38]: Did on https://github.com/aquasecurity/tracee/pull/4093

File: pkg/ebpf/tracee.go [hunk 3]
```diff
@@ -121,6 +121,11 @@ type Tracee struct {
 	policyManager *policyManager
 	// The dependencies of events used by Tracee
 	eventsDependencies *dependencies.Manager
+
+	// Ksymbols needed to be kept alive in table.
+	// This doens't mean they are required for tracee to function.
+	// TOOD: remove this in favor of dependency manager nodes
```
[the remark this answers is not in the document]
Comment (geyslan) [reply pkg/ebpf/tracee.go:127]: https://github.com/aquasecurity/tracee/pull/4093

**Comment by geyslan** [comment]
> To be replaced by #4092 that uses `environment` package instead of `libbpfgo/helpers`

The last one is https://github.com/aquasecurity/tracee/pull/4093. I'm closing this.

**Pull request closed without merging**